app/api_v1/resources.py

```diff
@@ -0,0 +1,369 @@
+from flask import abort, request, jsonify, url_for, g
+from ..models import User, Bucketlist, Item
+from flask_restful import Api, Resource, reqparse, fields, marshal
+from flask_sqlalchemy import SQLAlchemy
+from flask_httpauth import HTTPTokenAuth
+from app import db
+from ..serializers import bucketlist_serializer, item_serializer
+from base64 import b32encode
+
+auth = HTTPTokenAuth(scheme='Token')
+
+
+@auth.verify_token
+def verify_token(token):
+    """verifies token"""
+    user = User.verify_auth_token(token)
+    if not user:
+        return False
+    g.user = user
+    return True
+
+
+class RegisterApi(Resource):
+    def __init__(self):
+        self.parse = reqparse.RequestParser()
+        self.parse.add_argument(
+            'username', type=str, required=True, help='username not provided', location='form')
+        self.parse.add_argument(
+            'password', type=str, required=True, help='password not provided', location='form')
+        super(RegisterApi, self).__init__()
+
+    def post(self):
+        """
+        Register a new user given a username and a password
+
+        parameters:
+          -username
+          -password
+
+        """
+        args = self.parse.parse_args()
+        username = args['username']
+        password = args['password']
+        if User.query.filter_by(username=username).first() is not None:
+            return {'message': "User already exists"}, 409  # existing user
```

Comment:
This works but if you consider setting username as unique in your models then instead of checking if username exists, just try to add and commit within a try...except , it will throw an error which you can use to return the message in the except block. It reduces the operations carried out i.e checking then adding and committing.